### PR TITLE
tbump: init at 6.11.0

### LINKED ISSUE #315190: Package request: tbump
**Project description**

Tbump helps you bump the version of your project easily. It is a simple cli.

**Metadata**

* homepage URL: https://github.com/your-tools/tbump
* source URL: https://github.com/your-tools/tbump
* license: BSD 3-Clause
* platforms: Anything that run python

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by slashformotion**
## Description of changes

***Disclaimer: i am very much a noob at nix stuff***

This PR add tbump and one needed dependency.

fixes  #315190

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/development/python-modules/cli-ui/default.nix [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+  prePatch = ''
+    substituteInPlace pyproject.toml \
+        --replace 'tabulate = "^0.8.3"' 'tabulate = ">=0.8.3"'
+  '';
+
+  nativeBuildInputs = [
+    poetry-core
+  ];
+
+  propagatedBuildInputs = [
```
Comment (drupol) [inline pkgs/development/python-modules/cli-ui/default.nix:34]:
```suggestion
  dependencies = [
```

File: pkgs/development/tools/tbump/default.nix [hunk 2]
```diff
@@ -0,0 +1,48 @@
+{
```
Comment (drupol) [inline pkgs/development/tools/tbump/default.nix:1]: This can be moved to `pkgs/by-name`

File: pkgs/development/tools/tbump/default.nix [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+  disabled = pythonOlder "3.8";
+
+  src = fetchPypi {
+    inherit pname version;
+    hash = "sha256-OF5xDu3wqKb/lZzx6fPP0XyHNhcTL8DsX2Ka8MNVyHA=";
+  };
+  prePatch = ''
+    substituteInPlace pyproject.toml \
+        --replace 'tomlkit = "^0.11"' 'tomlkit = ">=0.11"'
+  '';
```
Comment (drupol) [inline pkgs/development/tools/tbump/default.nix:27]: Use also the `pythonRelaxDepsHook`

File: pkgs/development/tools/tbump/default.nix [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+  nativeBuildInputs = [
```
Comment (drupol) [inline pkgs/development/tools/tbump/default.nix:29]:
```suggestion
  build-system = [
```

File: pkgs/development/tools/tbump/default.nix [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    poetry-core
+  ];
+
+  propagatedBuildInputs = [
```
Comment (drupol) [inline pkgs/development/tools/tbump/default.nix:33]:
```suggestion
  dependencies = [
```

File: pkgs/development/tools/tbump/default.nix [hunk 6]
```diff
@@ -0,0 +1,48 @@
...
+    docopt
+    schema
+    packaging
+    poetry-core
```
Comment (drupol) [inline pkgs/development/tools/tbump/default.nix:37]: Is `poetry-core` required in here?

File: pkgs/development/tools/tbump/default.nix [hunk 7]
```diff
@@ -0,0 +1,48 @@
...
+    tomlkit
+    cli-ui
+  ];
+
+  meta = with lib; {
```
Comment (drupol) [inline pkgs/development/tools/tbump/default.nix:42]: Missing `meta.mainProgram`

File: pkgs/top-level/all-packages.nix [hunk 8]
```diff
@@ -19669,6 +19669,8 @@ with pkgs;
     inherit (darwin.apple_sdk.frameworks) Security;
   };
 
+  tbump = python3Packages.callPackage ../development/tools/tbump {};
+
```
Comment (drupol) [inline pkgs/top-level/all-packages.nix:19673]: This is not required if you move to `pkgs/by-name`.

**Review by drupol — CHANGES_REQUESTED** [review]
Hi !

Thanks for your first contribution.

I've left some comments, let me know if this is clear enough.

Feel free to ask for assistance!

File: pkgs/development/tools/tbump/default.nix (hunk 6 above)
Comment (slashformotion) [reply pkgs/development/tools/tbump/default.nix:37]: I removed it

File: pkgs/by-name/tb/tbump/package.nix [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+}:
+python3Packages.buildPythonApplication rec {
+  pname = "tbump";
+  version = "6.11.0";
+  pyproject = true;
+
+  disabled = python3Packages.pythonOlder "3.8";
+
+  src = fetchPypi {
+    inherit pname version;
```
Comment (drupol) [inline pkgs/by-name/tb/tbump/package.nix:14]: Please don't use this.

`repo = pname;` is nice for DRY but creates a binding that goes too far.

See further information about this here: https://github.com/nix-community/nixpkgs-lint/issues/21

File: pkgs/by-name/tb/tbump/package.nix [hunk 10]
```diff
@@ -0,0 +1,47 @@
...
+  
+  pythonRelaxDeps = [
+    "tomlkit"
+  ];
+
+  nativeBuildInputs = with python3Packages; [
+    pythonRelaxDepsHook
+  ];
+
+
```
Comment (drupol) [inline pkgs/by-name/tb/tbump/package.nix:26]: extra empty line

File: pkgs/development/python-modules/cli-ui/default.nix [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+
+  nativeBuildInputs = [
+    pythonRelaxDepsHook
+  ];
+
+  build-system = [
+    poetry-core
+  ];
+
+  dependencies = with python3Packages; [
```
Comment (drupol) [inline pkgs/development/python-modules/cli-ui/default.nix:34]:
```suggestion
  dependencies = [
```

Please pass those dependencies in the top level arguments. You should also remove `python3Packages` from the top level arguments.

**Review by drupol — CHANGES_REQUESTED** [review]
Nice improvements!

Don't forget to pass the new files through `nixfmt-rfc-style`.

File: pkgs/development/python-modules/cli-ui/default.nix (hunk 11 above)
Comment (drupol) [reply pkgs/development/python-modules/cli-ui/default.nix]: Sorry ignore me, this comment was for the other file actually.

**Comment by drupol** [comment]
@ofborg build tbump

**Review by drupol — APPROVED** [review]
LGTM! Congrats for your first PR... keep them coming!

Welcome to the Nix family! Be careful, it's addictive :)

**Pull request merged**